modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/PreviewJournalArticleContentDashboardItemVersionAction.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.content.dashboard.journal.internal.item.action;
+
+import com.liferay.content.dashboard.item.action.ContentDashboardItemVersionAction;
+import com.liferay.journal.constants.JournalPortletKeys;
+import com.liferay.journal.model.JournalArticle;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.portlet.LiferayPortletResponse;
+import com.liferay.portal.kernel.portlet.LiferayWindowState;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactory;
+import com.liferay.portal.kernel.portlet.url.builder.PortletURLBuilder;
+import com.liferay.portal.kernel.util.JavaConstants;
+import com.liferay.portal.kernel.util.PortalUtil;
+
+import java.util.Locale;
+
+import javax.portlet.PortletResponse;
+import javax.portlet.PortletURL;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Cristina González
+ */
+public class PreviewJournalArticleContentDashboardItemVersionAction
+	implements ContentDashboardItemVersionAction {
+
+	public PreviewJournalArticleContentDashboardItemVersionAction(
+		HttpServletRequest httpServletRequest, JournalArticle journalArticle,
+		Language language,
+		RequestBackedPortletURLFactory requestBackedPortletURLFactory) {
+
+		_httpServletRequest = httpServletRequest;
+		_journalArticle = journalArticle;
+		_language = language;
+		_requestBackedPortletURLFactory = requestBackedPortletURLFactory;
+	}
+
+	@Override
+	public String getIcon() {
+		return "view";
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(locale, "preview");
+	}
+
+	@Override
+	public String getName() {
+		return "preview";
+	}
+
+	public Type getType() {
+		return Type.NAVIGATE;
+	}
+
+	@Override
+	public String getURL() {
```

Comment:
I am seeing that maybe we are missing some stuffs here:

1. If the content has a display page template selected, we have to create the url to see the content with that display page template.
2. If the content does not have a template selected , we should not build an url to preview the content.

I saw this logic here: [JournalArticleActionDropdownItemsProvider](https://github.com/liferay/liferay-portal/blob/master/modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/servlet/taglib/util/JournalArticleActionDropdownItemsProvider.java)

Reply:
That is totally true, sorry, the main goal of this PR where for @beltranrengifo to review the type of the actions, as we was having a parallel discussion in https://github.com/liferay-tango/liferay-portal/pull/2968. I will mark it as draft.
 
 Sorry if I waste your time :(, because this Pr is not ready to be reviewed.

Reply:
Ok @cgoncas  understood , as I had seen the review request for that reason I reviewed it :) , thanks to you